### PR TITLE
support sepereate engine for taobao mini game

### PR DESCRIPTION
**Pull request by bofeng-song**
Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: pal/minigame/taobao_minigame.ts [hunk 1]
```diff
@@ -157,7 +157,7 @@ minigame.stopAccelerometer = function (res: any) {
 // #region SafeArea
 // It should be a value that is not multiplied by dpr
 minigame.getSafeArea = function () {
-    const systemInfo = minigame.getSystemInfoSync();
+    const systemInfo = minigame.getWindowInfoSync();
```
Comment (PPpro) [inline pal/minigame/taobao_minigame.ts]: this seems not to be a common method for all minigame platforms
please use `my.getWindowInfoSync()`

and add `declare const my: any;` before

**Comment by PPpro** [comment]
please fix the eslint error

File: platforms/minigame/common/engine/AssetManager.js [hunk 2]
```diff
@@ -18,13 +18,16 @@ const sys = cc.sys;
...
+}
 
 function downloadScript (url, options, onComplete) {
     if (REGEX.test(url)) {
         onComplete && onComplete(new Error('Can not load remote scripts'));
     } else {
-        if (sys.platform === sys.Platform.TAOBAO_MINI_GAME) {
-            __taobaoRequire(`../../../${url}`);
+        if (sys.platform === sys.Platform.TAOBAO_MINI_GAME) { //Can't load scripts dynamically on Taobao platform
+            require(`../../../${url}`);
```
Comment (minggo) [inline platforms/minigame/common/engine/AssetManager.js:30]: The comments say can not load, but it also invoke `require()`. It is wired.

File: platforms/minigame/common/engine/AssetManager.js [hunk 3]
```diff
@@ -253,8 +256,8 @@ function downloadBundle (nameOrUrl, options, onComplete) {
             js = `assets/${bundleName}/index.${suffix}js`;
         }
 
-        if (sys.platform === sys.Platform.TAOBAO_MINI_GAME) {
-            __taobaoRequire(js);
+        if (sys.platform === sys.Platform.TAOBAO_MINI_GAME) { // Can't load scripts dynamically on Taobao platform
+            require(js);
```
Comment (minggo) [inline platforms/minigame/common/engine/AssetManager.js:260]: The same as above.

**Review by minggo — APPROVED** [review]

**Review by PPpro — APPROVED** [review]

**Pull request merged**